profile/profile_test.go

```diff
@@ -997,7 +1075,7 @@ func locationHash(s *Sample) string {
 	var tb string
 	for _, l := range s.Location {
 		for _, ln := range l.Line {
-			tb = tb + fmt.Sprintf("%s:%d@%d ", ln.Function.Name, ln.Line, l.Address)
+			tb = tb + fmt.Sprintf("%s:%d:%d@%d ", ln.Function.Name, ln.Line, ln.Column, l.Address)
```

Comment:
When columns are present, are they 1-based? If yes, maybe we can omit the ":0" part when the column is zero? This would make the `pprof -raw` output less chatty for the case when the column information is missing.

The downside is that if the column information is only partially present then the `-raw` output would partially have and partially haven't the column number which can be obscure to the user.

Reply:
Yes, column numbers are 1-based when available (see https://dwarfstd.org/doc/DWARF5.pdf, Sec 6.2.2 pg 157). The value 0 is reserved to indicate that a statement begins at the “left edge” of the line, but implementations don't use this as far as I know. 

IMO it's clearer to retain the value, an additional zero per line is not too chatty. Happy to reconsider if you feel strongly.

Reply:
In the end I chose to suppress the zero column numbers while reporting to minimize updates to the tests.

Reply:
I don't see it being suppressed, am I missing something?

Reply:
This is the test-only locationHash helper. Here I think we need the extra resolution here. My comment about suppressing the column number is implemented in `graph.go`, see L178-181 which updated the `NameComponents` method. 

```
s := fmt.Sprintf("%s:%d", i.File, i.Lineno)
if i.Columnno != 0 {
	s += fmt.Sprintf(":%d", i.Columnno)
}
```